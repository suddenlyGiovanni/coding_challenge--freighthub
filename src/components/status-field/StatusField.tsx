import * as React from 'react'
import TextField from '@material-ui/core/TextField'
import MenuItem from '@material-ui/core/MenuItem'
import { makeStyles, createStyles } from '@material-ui/core/styles'

const status = [
  { value: 'ALL', label: 'All' },
  { value: 'ACTIVE', label: 'In Progress' },
  { value: 'COMPLETED', label: 'Completed' },
]

const useStyles = makeStyles(
  createStyles({
    textField: {
      width: 200,
    },
    menu: {
      width: 200,
    },
  })
)

type State = 'ACTIVE' | 'COMPLETED' | 'ALL'

interface Props {
  onChange: (value: State) => void
  initialValue?: State
}

export const StatusField: React.FC<Props> = ({
  onChange,
  initialValue = 'ALL',
}) => {
  const classes = useStyles()
  const [value, setValue] = React.useState(initialValue)

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const _value = event.target.value as State
    setValue(_value)
    onChange(_value)
  }

  return (
    <TextField
      id="status"
      className={classes.textField}
      select
      label="Status"
      value={value}
      onChange={handleChange}
      margin="none"
      SelectProps={{
        MenuProps: {
          className: classes.menu,
        },
      }}
    >
      {status.map(option => (
        <MenuItem key={option.value} value={option.value}>
          {option.label}
        </MenuItem>
      ))}
    </TextField>
  )
}
