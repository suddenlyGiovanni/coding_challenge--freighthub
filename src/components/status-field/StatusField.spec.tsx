import React from 'react'
import { render } from '@testing-library/react'

import { StatusField } from './StatusField'

describe('StatusField', () => {
  it.todo('calls `onChange` with the new status when selection is changed')
  // it is really complicate to simulate a change on the material ui select element.

  test('snapshot', () => {
    const { container } = render(<StatusField onChange={() => {}} />)
    expect(container).toMatchSnapshot()
  })
})
