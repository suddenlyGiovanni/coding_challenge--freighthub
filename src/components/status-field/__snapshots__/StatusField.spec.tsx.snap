// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusField snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
    data-testid="status-select"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="status"
    >
      Status
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        class="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          All
        </div>
        <input
          id="status"
          type="hidden"
          value="ALL"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
